### ISSUE TITLE
CRA typescript addon-docs all descriptions unknown, missing props from interface

### ISSUE THREAD

**Issue by lockykeaney** [storybookjs/storybook#10158]
**Describe the bug**
The `addon-docs` is not reading my components TypeScript interface and populating the prop table correctly. All descriptions are unknown, some props are missing completely. Some components don't find the props at all.

**Expected behavior**
Type Interface to be read and populate the prop table with prop name and description.

**Screenshots**
<img width="1048" alt="Screen Shot 2020-03-19 at 10 32 20 am" src="https://user-images.githubusercontent.com/12781843/77016923-277df600-69cd-11ea-9881-159d1940a12b.png">

**Code snippets**
Story
```
import React from 'react';
import Button from './Button';

export default {
  title: 'Button',
  component: Button,
};

export const Default = () => (
  <Button variant="primary" onClick={() => console.log('Click')}>
    Default Button
  </Button>
);
```

Component
```
import React from 'react';

import { StyledButton, ButtonVariants } from './buttonStyle';

import { Icons, FlexContainer } from 'components';

export interface ButtonProps {
  /** Label for the Button to display */
  children: string;
  /** Function when the button is clicked */
  onClick: () => void;
  /** The different button styles available */
  variant?: ButtonVariants;
  /** Type of button. Submit is for the FormWrapper */
  type?: 'submit' | 'reset' | 'button';
  /** Disabled the button  */
  disabled?: boolean;
  /** If true, button will fill container. If false, button will fit content */
  stretch?: boolean;
  /** Name of the Icon to use. Will inherit color. */
  icon?: any;
}

/** The Basic Button */

const Button: React.FC<ButtonProps> = ({
  children,
  onClick,
  variant = 'primary',
  type = 'button',
  disabled = false,
  stretch = false,
  icon,
}) => {
  return (
    <StyledButton
      variant={variant}
      onClick={onClick}
      disabled={disabled}
      type={type}
      stretch={stretch}>
      <FlexContainer centered>
        {icon && (
          <span style={{ marginRight: 10 }}>
            <Icons icon={icon} />
          </span>
        )}
        {children}
      </FlexContainer>
    </StyledButton>
  );
};

export default Button;
```

main.js
```
module.exports = {
  stories: ['../src/**/*.stories.(tsx|mdx)'],
  addons: [
    '@storybook/addon-viewport/register',
    '@storybook/addon-actions',
    'storybook-formik/register',
    '@storybook/theming',
    '@storybook/preset-typescript',
    '@storybook/preset-create-react-app',
    '@storybook/addon-docs',
  ],
};
```

storybook packages
```
"@storybook/addon-actions": "^5.3.14",
    "@storybook/addon-centered": "^5.3.14",
    "@storybook/addon-docs": "^5.3.17",
    "@storybook/addon-links": "^5.3.14",
    "@storybook/addon-viewport": "^5.3.14",
    "@storybook/addons": "^5.3.14",
    "@storybook/preset-create-react-app": "^2.0.0",
    "@storybook/preset-typescript": "^2.1.0",
    "@storybook/react": "^5.3.14",
```

**Additional context**
I have tried a few upgrades to packages, but there seems to be an unfixable error with ts-loader where I get `Line: 0 name is undefined` or something along those lines. Not sure if that's a problem on your end but I highly doubt. Would prefer to not have to upgrade too much.

I tried converting to MDX for documentation but that had a whole lot more errors with the MDX loader.

**Comment by shilman** [comment]
@lockykeaney Because you're using `preset-create-react-app` you're running up against https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#reactfc-interfaces. Please try updating your code:

```ts
const Button: React.FC<ButtonProps> = ({
  children,
  onClick,
  variant = 'primary',
  type = 'button',
  disabled = false,
  stretch = false,
  icon,
}: ButtonProps) => ...
```

See migration instructions for more info: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#react-prop-tables-with-typescript

**Comment by lockykeaney** [comment]
Thanks @shilman, that work around is fine for me at the moment 👍. And I updated to 2.1.1.

The Prop table is now populated, but the description does not show the values allowed, just returns `ButtonVariants` or `union`, but I don't think this is an issue on your side,

**Comment by shilman** [comment]
@lockykeaney glad to hear it's sort of working. it is an issue in Storybook -- the type rendering code needs to be updated to the new setup. coming soon!

**Comment by shilman** [comment]
We've just released zero-config typescript support in 6.0-beta. Please upgrade and test it!

- [Release 6.0.0-beta.12+](https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.12)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#zero-config-typescript)
- [Updated docs](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/typescript-config/index.md)

Thanks for your help and support getting this stable for release!
